src/wallet/test/wallet_tests.cpp

```diff
@@ -677,11 +677,11 @@ BOOST_FIXTURE_TEST_CASE(ListCoins, ListCoinsTestingSetup)
 
     // Lock both coins. Confirm number of available coins drops to 0.
     std::vector<COutput> available;
+    LOCK2(cs_main, wallet->cs_wallet);
```

Comment:
This locks the cs_main and cs_wallet for the entire rest of this test, which is somewhat awkward scoping-wise, is that intentional?

Reply:
Nothing else happens and below the lock is again required so I've decided for the minimal diff. I can lock only until `ListCoins` below if you prefer.

Reply:
Yes please, For readability I prefer locks to be taken at the beginning of a block.

Reply:
Done, 2 blocks instead of one.